mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -228,8 +234,51 @@ func (c *Config) synchronousTimeout() time.Duration {
 	return c.RequestTimeout
 }
 
-func (c *Config) getTokenSource(clientScopes []string) (oauth2.TokenSource, error) {
-	creds, err := c.GetCredentials(clientScopes)
+// Print Identities executing terraform API Calls.
+func (c *Config) logGoogleIdentities() error {
+	if c.ImpersonateServiceAccount == "" {
+
+		email, err := GetCurrentUserEmail(c, c.userAgent)
```

Comment:
Ah, whoops, I missed that `terraform-google-conversion` is failing to build. Can we move this function into utils.go so it's available there?

Reply:
It is crashing when you move the function. I pushed that change.

```
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccComputeZones_basic -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeZones_basic
=== PAUSE TestAccComputeZones_basic
=== CONT  TestAccComputeZones_basic
2021/05/13 19:38:10 [INFO] Authenticating using DefaultClient...
2021/05/13 19:38:10 [INFO]   -- Scopes: [https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/cloud-identity https://www.googleapis.com/auth/ndev.clouddns.readwrite https://www.googleapis.com/auth/devstorage.full_control https://www.googleapis.com/auth/userinfo.email]
2021/05/13 19:38:10 [DEBUG] Waiting for state to become: [success]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x132bac2]

goroutine 133 [running]:
net/http.(*Client).deadline(0x0, 0x20, 0x20, 0x3c57760)
	/usr/local/Cellar/go/1.15.5/libexec/src/net/http/client.go:189 +0x22
net/http.(*Client).do(0x0, 0xc000e24500, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.15.5/libexec/src/net/http/client.go:604 +0x2bd
net/http.(*Client).Do(...)
	/usr/local/Cellar/go/1.15.5/libexec/src/net/http/client.go:586
github.com/hashicorp/terraform-provider-google-beta/google-beta.sendRequestWithTimeout.func1(0xc000cf5e4c, 0xc0015bea80)
	/Users/REDACTED/go/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/transport.go:83 +0x252
github.com/hashicorp/terraform-provider-google-beta/google-beta.retryTimeDuration.func1(0x1096d65)
	/Users/REDACTED/go/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/retry_utils.go:21 +0x32
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/REDACTED/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/resource/wait.go:27 +0x5b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1(0xc000ffc3c0, 0xc000a42230, 0xc0007846c0, 0xc000a79ec0, 0xc000f18370, 0xc000f18368)
	/Users/REDACTED/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/resource/state.go:110 +0x344
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	/Users/REDACTED/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/resource/state.go:83 +0x1c6
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	7.557s
FAIL

```

Reply:
Ah, I should have stated explicitly the function- I forgot that GH doesn't show the highlighted text. `GetCurrentUserEmail` is the one we need to move, not `logGoogleIdentities`